src/System.Private.ServiceModel/src/System/ServiceModel/Description/TypeLoader.cs

```diff
@@ -499,6 +495,56 @@ private KeyedByTypeCollection<IOperationBehavior> GetIOperationBehaviorAttribute
             }
         }
 
+        // Given a MethodInfo which could be from any type, return a MethodInfo declared by the given type 
+        // that matches in name and parameter types.  Null is returned if there is no match.
+        private static MethodInfo GetCorrespondingMethodFromType(Type type, MethodInfo methodInfo)
+        {
+            Contract.Assert(type != null);
+            Contract.Assert(methodInfo != null);
+
+            if (methodInfo.DeclaringType == type)
+            {
+                return methodInfo;
+            }
+
+            MethodInfo matchingMethod = type.GetTypeInfo().DeclaredMethods.SingleOrDefault(m => MethodsMatch(m, methodInfo));
+            return matchingMethod;
+        }
+
+        // Returns true if the given methods match in name and parameter types
+        private static bool MethodsMatch(MethodInfo method1, MethodInfo method2)
+        {
+            Contract.Assert(method1 != null);
+            Contract.Assert(method2 != null);
+
+            if (method1 == method2)
+            {
+                return true;
+            }
+
+            if (!String.Equals(method1.Name, method2.Name, StringComparison.Ordinal))
+            {
+                return false;
+            }
+
+            ParameterInfo[] parameters1 = method1.GetParameters();
+            ParameterInfo[] parameters2 = method2.GetParameters();
+            if (parameters1.Length != parameters2.Length)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < parameters1.Length; ++i)
+            {
+                if (parameters1[0].ParameterType != parameters2[i].ParameterType)
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
```

Comment:
No, async tells the compiler to generate code for a state machine for async continuations from await calls within the method body and automatically create a Task to be returned. It has no effect on the method signature at all. If you decompile an async method, the only way you know that the method was declared async is that the code is almost impossible to follow :smile: